README.md

```diff
@@ -65,9 +65,23 @@ This syntax can be used with any decorator type, and is used in cases where addi
 
 ## Detailed Design
 
-In general, decorators receive two parameters:
+The three steps of decorator evaluation:
+
+1. Decorator expressions (the thing after the `@`) are *evaluated* interspersed with computed property names.
+1. Decorators are *called* (as functions) during class definition, after the methods have been evaluated but before the constructor and prototype have been put together.
+1. Decorators are *applied* (mutating the constructor and prototype) all at once, after all of them have been called.
```

Comment:
This list only applies to instance method/property decorators, no?

Reply:
Correct, it does not apply to class decorators